lib/ipam/ipam_block.go

```diff
@@ -21,56 +21,49 @@ import (
 	"net"
 	"reflect"
 
+	"github.com/projectcalico/libcalico-go/lib/apis/v3"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/projectcalico/libcalico-go/lib/backend/model"
 	cnet "github.com/projectcalico/libcalico-go/lib/net"
 )
 
-const (
-	blockSize = 64
-)
-
-type ipVersion struct {
-	Number            int
-	TotalBits         int
-	BlockPrefixLength int
-	BlockPrefixMask   net.IPMask
-}
-
-var ipv4 ipVersion = ipVersion{
-	Number:            4,
-	TotalBits:         32,
-	BlockPrefixLength: 26,
-	BlockPrefixMask:   net.CIDRMask(26, 32),
-}
-
-var ipv6 ipVersion = ipVersion{
-	Number:            6,
-	TotalBits:         128,
-	BlockPrefixLength: 122,
-	BlockPrefixMask:   net.CIDRMask(122, 128),
-}
-
 // Wrap the backend AllocationBlock struct so that we can
 // attach methods to it.
 type allocationBlock struct {
 	*model.AllocationBlock
+	numAddresses int
 }
 
-func newBlock(cidr cnet.IPNet) allocationBlock {
+func newBlock(cidr cnet.IPNet, blockSize int) allocationBlock {
```

Comment:
I'm a bit confused why we need both `cidr` and `blockSize` here, especially since it looks like all the calling code calculates `blockSize` using `cidr.Mask.Size()`